lib/pagetience.rb

```diff
@@ -65,21 +85,17 @@ def wait_for_required_elements
     end
   end
 
-  module ClassMethods
-    def required(*elements)
-      elements.keep_if { |e| e.is_a? Symbol }
-      define_method('_required_elements') do
-        elements
-      end
-    end
+  private
 
-    def waiting(timeout, polling=1)
-      define_method('_waiting_timeout') do
-        timeout
-      end
-      define_method('_waiting_polling') do
-        polling
+  def determine_platform
+    # Search the ancestors
+    self.class.ancestors.find { |a| SUPPORTED_PLATFORMS[:ancestors].include? a }
+    SUPPORTED_PLATFORMS[:ancestors].each do |p|
+      if self.class.ancestors.include? p[:module]
+        @element_platform = p[:platform].new self
       end
     end
```

Comment:
Should move this out to its own method since there will be other ways of "finding" the platform.